Adds a config option to filter the results.

rossjones opened the pull request:
New configuration options 'organizations' which should be a list
of organization names.  If set each dataset is checked to ensure
its ['organization']['name'] is in the list - if not, it is ignored

rossjones committed:
Adds a config option to filter the results.

New configuration options 'organizations' which should be a list
of organization names.  If set each dataset is checked to ensure
its ['organization']['name'] is in the list - if not, it is ignored

davidread commented:
Looks good. Just one thing, how about renaming the config option from 'organizations' to 'organizations_filter'?

While you're at it can you add the opposite option 'organizations_filter_out' which excludes the ones you list? I think that's probably what we need for harvesting the London Datastore - we want their datasets apart from boroughs which already harvest with us directly, and where there is clear overlap such as ONS and DfE. There are probably plenty in a grey area where we may end up with duplicates, but I think we want to assume we'll harvest stuff and manually exclude stuff as we find the problems. (We also need to have a mapping of their organization names to ours, but that's a separate task http://redmine.dguteam.org.uk/issues/1856 )

rossjones committed:
Refactored config to allow include/exclude filters for remote
organizations

Now it is possible to ask for one of the following filters when
harvesting:

1. Retrieve only those datasets in a specific list of remote organizations

2. Retrieve all datasets except those in a specific list of remote
organizations.

rossjones commented:
Added `organizations_filter_include` and `organizations_filter_exclude`.

Having trouble with tests at the moment though.

davidread commented on README.rst:
```diff
@@ -179,6 +179,20 @@ field. The currently supported configuration options are:
     packages regardless of the modification date. Default is False.
 
+*   organizations_filter_include: This configuration option allows you to specify
+    a list of remote organization short names. If this property has a
```
I don't think we've referred to "short name" before. How about calling it "organization name" but give an example: "(e.g. "arkansas-gov" is the name for organization http://catalog.data.gov/organization/arkansas-gov )"?

davidread commented on ckanext/harvest/harvesters/ckanharvester.py:
```diff
@@ -269,6 +278,19 @@ def fetch_stage(self, harvest_object):
+        if org_name:
+            if self.organizations_include:
+                if not org_name in self.organizations_include:
+                    log.debug('Skipping dataset %s as not in organizations_filter_include' % dataset['name'])
```
maybe add org_name to the debug too?

rossjones committed:
Fixes tests and small text changes in logging

rossjones commented:
Will try and get this tested ;)

rossjones commented:
Tested (manually) against LDM by 'including' Leeds City Council ... got only those datasets.  Then with exclude (for a remote org with one dataset) the harvest completed and was correctly missing the excluded org's datasets.
